### PR TITLE
[release-automation] More flexibility on the pipeline

### PR DESCRIPTION
**Pull request by khluu**
- Only upload `linux_x86_64`, `mac x86_64`, and `mac arm64` wheels unless it's a full platform release.
- Each of the wheel sanity check step has its own block step.
- Uploading wheels to PyPI no longer requires all sanity checks to pass. This is up for release manager to judge whether the wheels are ready.
- Add key for each of the step. This is to support automation that checks for BK job status (it's better to retrieve jobs from BK and identify with key than job label)

### CODE REVIEW

**Review by aslonnie — APPROVED** [review]

File: ci/ray_ci/automation/ray_wheels_lib.py [hunk 1]
```diff
@@ -82,8 +92,9 @@ def download_ray_wheels_from_s3(
         directory_path: The directory to download the wheels to.
     """
     full_directory_path = os.path.join(bazel_workspace_dir, directory_path)
-
-    wheels = _get_wheel_names(ray_version)
+    _, minor_version, _ = ray_version.split(".")
```
Comment (aslonnie) [inline ci/ray_ci/automation/ray_wheels_lib.py:95]: this might not always be 3 parts , maybe just take 2nd part without assuming it has 3 parts?

Comment (khluu) [reply ci/ray_ci/automation/ray_wheels_lib.py]: oh ya I forgot about cases like `0.0.1.post1`. Updated to just grab the 2nd part

File: ci/ray_ci/automation/test_ray_wheels_lib.py [hunk 2]
```diff
@@ -134,10 +162,42 @@ def test_download_ray_wheels_from_s3(
...
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        download_ray_wheels_from_s3(
+            commit_hash=commit_hash,
+            ray_version=ray_version,
+            directory_path=tmp_dir,
         )
 
-        mock_get_wheel_names.assert_called_with(ray_version)
+        mock_get_wheel_names.assert_called_with(ray_version, False)
```
Comment (aslonnie) [inline ci/ray_ci/automation/test_ray_wheels_lib.py:200]: nit: add `full_platform=False` for readability?

it is actually often not recommended to use a boolean as an flag arg. a string or an enum is often more readable.

**Review by aslonnie — APPROVED** [review]

File: ci/ray_ci/automation/test_ray_wheels_lib.py (hunk 2 above)
Comment (khluu) [reply ci/ray_ci/automation/test_ray_wheels_lib.py]: done

**Pull request merged**